### ISSUE TITLE
More resource exhaustion issues with the docker callback solution.  

### ISSUE THREAD

**Issue by TerryE** [TerryE/docker-buildhub#10]
My last four issues: #5, #6, #8 and #9 have all been related to the callback and logging implementation.  I am still having exhaustion issues so there is still something wrong here.  The `hk` instance runs a crontab script does which currently has various script entries local to `hk` and the following entries which use the `host-callback.sh` script to make calls to housekeeping in other docker services:
-   `php` **task** callback every minute.
-   The nightly logrotate makes flush-log requests to `httpd` , `php` and `sshd`.   (`mysql` uses docker logging)
-  `httpd` **cerbot** check weekly.

 `host-callback.sh`  writes to a pipe which is read by `docker-callback-reader` and uses the message parameters to do a docker callback to the relevant service's `docker-service-callback.sh` script.   `docker-callback-reader`  is started in my crontab as follows:
```` bash
@reboot   prod/bin/docker-callback-reader > /dev/null
@hourly    prod/bin/docker-callback-reader > /dev/null
````

The `@hourly` entry is just for resilience  to restart should the script fail.  I've got two underlying issues:

The main one is that I only want one instance of the callback reader running, so the script includes multi-instance detection to allow any second instance to exit cleanly.  I am using a standard `flock` pattern for this but am using the logging file for this.  _Stupid_, as multiple processes can write to this stdout file and a write `flock` will block these.  I need to use a dedicated `.lock` file here instead.  This will prevent process backlog from output to the file.  Durrrh.
 
 Secondly, I don't want the docker exec calls to block each other so they need to be run detached.  I am currently using a CLI level detach (the `&` operator), but of course this should be done within `docker` using the `-d` flag.  Either way the php `task` function usually completes in under a second but some invocations can run for well over a minute.  I am not sure how elegantly the php task function handles multiple instances.  Better to add an flock single instance logic here as well.

**Comment by TerryE** [comment]
Fixed by 6f07f6e.

**Comment by mikesharp01** [comment]
Well done Terry...

Mike
